src/bio/smd/smd_pool.c

```diff
@@ -86,12 +87,17 @@ pool_add_tgt(uuid_t pool_id, uint32_t tgt_id, uint64_t blob_id, char *table_name
 		pool.sp_tgts[pool.sp_tgt_cnt] = tgt_id;
 		pool.sp_blobs[pool.sp_tgt_cnt] = blob_id;
 		pool.sp_tgt_cnt += 1;
+		if (!strcmp(table_name, TABLE_POOLS[SMD_DEV_TYPE_META]))
```

Comment:
To avoid possible bugs with unterminated strings, we should be using the `n` versions of these string functions, e.g. `strncmp`, with some reasonable maximum length.

Reply:
Both the arguments are static strings used with the module and are of variable sizes. Hence used strcmp().

Reply:
Fixed this to use strncmp as per suggestion from @jolivier23 as well.